app-modules/engagement/src/Actions/FindEngagementResponseSender.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace Assist\Engagement\Actions;
+
+use Assist\Prospect\Models\Prospect;
+use Assist\AssistDataModel\Models\Student;
+use Assist\Engagement\Exceptions\UnknownEngagementSenderException;
+
+class FindEngagementResponseSender
+{
+    public function __invoke(string $phoneNumber): Student|Prospect
+    {
+        // Student currently takes priority, but determine if we potentially want to store this response
+        // For *all* potential matches instead of just a singular result.
```

Comment:
```suggestion
        // For *all* potential matches instead of just a singular result.
        // TODO: Make use of shared Student/Prospect implementation
```

I am adding this TODO here as I would like to discuss with you adding an interface, called `Entity` or `StudentEntity`.
It would enforce methods such as `getName()` and `getMobile()` so we can ensure that everywhere that we need to get the same value from either Model we know that no changes in the system would require changing what that column name is.